Typedef crash with duplicated invalid type

```c
typedef float *invalid1 __attribute__((vector_size(8)));
typedef float invalid1;
```

```
panic: reached unreachable code
/Users/ehaas/source/arocc/src/aro/Type.zig:2350:21: 0x102d8db3a in fromType (arocc)
            else => unreachable,
                    ^
/Users/ehaas/source/arocc/src/aro/Type.zig:1915:34: 0x102e0d32c in typeSpec (arocc)
        const new_spec = fromType(typedef_ty);
                                 ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:1465:27: 0x102ddd43c in declSpec (arocc)
        if (try p.typeSpec(&spec)) continue;
                          ^
```

--- a/src/aro/SymbolStack.zig
+++ b/src/aro/SymbolStack.zig
@@ -178,9 +178,11 @@ pub fn defineTypedef(
     if (s.get(name, .vars)) |prev| {
         switch (prev.kind) {
             .typedef => {
-                if (!ty.eql(prev.ty, p.comp, true)) {
-                    try p.errStr(.redefinition_of_typedef, tok, try p.typePairStrExtra(ty, " vs ", prev.ty));
-                    if (prev.tok != 0) try p.errTok(.previous_definition, prev.tok);
+                if (!prev.ty.is(.invalid)) {
+                    if (!ty.eql(prev.ty, p.comp, true)) {
+                        try p.errStr(.redefinition_of_typedef, tok, try p.typePairStrExtra(ty, " vs ", prev.ty));
+                        if (prev.tok != 0) try p.errTok(.previous_definition, prev.tok);
+                    }
                 }
             },
             .enumeration, .decl, .def, .constexpr => {

--- a/src/aro/Type.zig
+++ b/src/aro/Type.zig
@@ -1899,6 +1899,7 @@ pub const Builder = struct {
 
     /// Try to combine type from typedef, returns true if successful.
     pub fn combineTypedef(b: *Builder, p: *Parser, typedef_ty: Type, name_tok: TokenIndex) bool {
+        if (typedef_ty.is(.invalid)) return false;
         b.error_on_invalid = true;
         defer b.error_on_invalid = false;
 

--- a/test/cases/redefine invalid typedef.c
+++ b/test/cases/redefine invalid typedef.c
@@ -1,0 +1,5 @@
+typedef float *invalid1 __attribute__((vector_size(8)));
+typedef float invalid1;
+
+#define EXPECTED_ERRORS "redefine invalid typedef.c:1:40: error: invalid vector element type 'float *'" \
+

